ocaml/xapi/xapi_vm.ml

```diff
@@ -1444,6 +1444,31 @@ let set_appliance ~__context ~self ~value =
   (* Update the VM's allowed operations - this will update the new appliance's operations, if valid. *)
   update_allowed_operations ~__context ~self
 
+let set_groups ~__context ~self ~value =
+  let assert_only_one_placement_group groups singleton_placements =
```

Comment:
For the name "assert_only_one_placement_group", if an `affinity` group is added later, I think a VM should be able to join an `anti-affinity` group and `affinity` group at the same time?